lib/puppet/resource.rb

```diff
@@ -11,9 +11,12 @@ class Puppet::Resource
   include Puppet::Util::Tagging
 
   include Enumerable
-  attr_accessor :file, :line, :catalog, :exported, :virtual, :validate_parameters, :strict
+  attr_accessor :file, :line, :catalog, :exported, :virtual, :strict
   attr_reader :type, :title
 
+  # @deprecated
```

Comment:
@kylog @joshcooper - the construct for attr_accessor for :validate_parameters looks odd - can one of you help shed some light on what it is used for, or if it is dead code. I an imagine that it is used to conveniently get empty methods that implementations of a resource type can supply extra validation for, but I am unsure.